File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        values = []
        curr = head
        while curr:
            values.append(curr.val)
            curr = curr.next
        
        i = 0
        j = len(values) - 1
        while i < j:
            if values[i] != values[j]:
                return False
            i += 1
            j -= 1
        
        return True

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
  // convert linked list to an array of nums
  let nums = [];
  let curr = head;
  while (curr) {
    nums.push(curr.val);
    curr = curr.next;
  }
  
  // run algo to check for palindrome - two pointers
  let l = 0, r = nums.length - 1;
  while (l < r) {
    if (nums[l] !== nums[r]) return false;
    l++;
    r--;
  }
  
  return true;
};